Libraries/WNContainers/inc/WNRangePartition.h

```diff
@@ -307,8 +336,49 @@ inline void range_partition<NodeAllocator, T_SIZE>::release_interval(
     m_used_space -= token.size();
   } else {
     m_used_space -= token.size();
-    token.node->m_next_free = m_free_list;
-    m_free_list = token.node;
+    auto* n = token.node->merge_left_and_free();
+    if (n) {
+      m_node_allocator.free_node(token.node);
+      if (n->m_next && n->m_next->m_next_free) {
+        // We want to merge right, but this is the slow path.
+        // We have to remove this node from the free-list.
+        auto* free = m_free_list;
```

Comment:
I thought we were gonna try not to use `auto` that much. I guess in this case it doesn't really matter what the type is. I just find that `auto` makes code less readable.

Reply:
Fixed